### PR TITLE
Player Join Effect

### PR DESCRIPTION
**Pull request by audiodk**
Added firework explosion on join.  Can be turned off in the config.  Not sure how to make it add new things to config though. Will try to learn that now.

### CODE REVIEW

File: src/main/resources/config.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ enable-whois-on-player-join: true
 enable-whois-on-command: true
 whois-admin-permission-name: ww.admin
 whois-no-match-found-message: "&4&lNo matching player found with your whois query"
+whois-player-on-first-join-fireworks: false
```
Comment (Xinecraft) [inline src/main/resources/config.yml:24]: Plz rename this variable to "firework-on-join". and add it to the last of the config.yml

File: src/main/java/com/xinecraft/Minetrax.java [hunk 2]
```diff
@@ -136,6 +138,7 @@ public void onEnable()
         whoisNoMatchFoundMessage = this.getConfig().getString("whois-no-match-found-message");
         whoisPlayerOnJoinMessage = this.getConfig().getStringList("whois-player-on-join-message");
         whoisPlayerOnFirstJoinMessage = this.getConfig().getStringList("whois-player-on-first-join-message");
+        whoisPlayerOnJoinFireworks = this.getConfig().getBoolean("whois-player-on-first-join-fireworks");
```
Comment (Xinecraft) [inline src/main/java/com/xinecraft/Minetrax.java]: rename this accordingly

File: src/main/java/com/xinecraft/Minetrax.java [hunk 3]
```diff
@@ -74,6 +74,8 @@ public final class Minetrax extends JavaPlugin implements Listener {
     @Getter
     private List<String> whoisPlayerOnCommandMessage;
     @Getter
+    private Boolean whoisPlayerOnJoinFireworks;
```
Comment (Xinecraft) [inline src/main/java/com/xinecraft/Minetrax.java]: rename to isFireworkOnPlayerJoin

File: src/main/resources/config.yml (hunk 1 above)
Comment (audiodk) [inline src/main/resources/config.yml]: okay give me a minute

Comment (Xinecraft) [inline src/main/resources/config.yml]: Plz rename this variable to "enable-firework-on-player-join". and add it to the last of the config.yml

**Review by Xinecraft — APPROVED** [review]

File: src/main/java/com/xinecraft/Minetrax.java [hunk 4]
```diff
@@ -159,6 +165,13 @@ public void onEnable()
             return;
         }
 
+        //Disable plugin if fireworks on first join and on join are both set to true
```
Comment (Xinecraft) [inline src/main/java/com/xinecraft/Minetrax.java]: Why to disable whole plugin for some misconfig which is not related.

File: src/main/java/com/xinecraft/listeners/PlayerJoinLeaveListener.java [hunk 5]
```diff
@@ -32,6 +44,36 @@ public void onPlayerJoin(PlayerJoinEvent event) {
 
         // perform whois & add player to list of linkedPlayers hashmap
         this.broadcastWhoisForPlayer(event.getPlayer());
+
+        // fireworks effect when player joins
+        if (Minetrax.getPlugin().getIsFireworkOnPlayerJoin()) {
+            Firework fw = player.getWorld().spawn(player.getLocation(), Firework.class);
```
Comment (Xinecraft) [inline src/main/java/com/xinecraft/listeners/PlayerJoinLeaveListener.java]: Code duplications when firing fireworks.
Can be changed to function.

**Review by Xinecraft — CHANGES_REQUESTED** [review]

File: src/main/java/com/xinecraft/listeners/PlayerJoinLeaveListener.java (hunk 5 above)
Comment (audiodk) [reply src/main/java/com/xinecraft/listeners/PlayerJoinLeaveListener.java]: Will work on this later today

**Pull request merged**